HTML&CSS Homework

oladelmar opened the pull request.

Commit:
Create page layout

Commit:
Modify readme.md

IvanCeban commented on index.html:
```diff
@@ -0,0 +1,84 @@
+        <title>VueJS Academy &mdash; MAfonina's HTML&amp;CSS homework</title>
+        
+        <link rel="apple-touch-icon" sizes="180x180" href="/favicons/apple-touch-icon.png">
+        <link rel="icon" type="image/png" sizes="32x32" href="/favicons/favicon-32x32.png">
```
Thanks for using favicon!

IvanCeban commented on index.html:
```diff
@@ -0,0 +1,84 @@
+    <body class="container">
+       
+        <div class="logo">
+           <svg class="logo__image">
```
using svg is a big plus!

IvanCeban commented on index.html:
```diff
@@ -0,0 +1,84 @@
+        
+        <header class="header">
+            <nav class="main-navigation">
+                <ul class="main-navigation__list">
```
Thank you for using BEM!

IvanCeban commented on index.html:
```diff
@@ -0,0 +1,84 @@
+        <meta name="msapplication-config" content="/favicons/browserconfig.xml">
+        <meta name="theme-color" content="#ffffff">
+    </head>
+    <body class="container">
```
Avoid adding layout styles to body. Better create a separate div with class "container" and place all content in it. Body is used only for very general styling like global font, box-sizing and so on.

IvanCeban commented on index.html:
```diff
@@ -0,0 +1,84 @@
+            <span>Log in</span>
+        </header>
+        
+        <aside class="sidebar">
```
Very good idea to make the sidebar toggleable!
Here are some suggestions to make it better:
1) You actually don't need to add background as a separate element (you can remove this div). Better set background color and all other styles to .side-navigation
2) Move your checkbox out of .site-navigation to make them siblings (in this case you could set styles to .side-navigation through checkbox)

IvanCeban commented on index.html:
```diff
@@ -0,0 +1,84 @@
+            </div>
+        </div>
+        
+        <footer class="footer">
```
Thanks for using semantic tags!

IvanCeban commented on package.json:
```diff
@@ -0,0 +1,23 @@
+{
```
Thanks for using npm and scss!!!

IvanCeban commented on sass/main.scss:
```diff
@@ -0,0 +1,8 @@
+@import "base";
```
Great division of scss files!

IvanCeban commented on sass/_components.scss:
```diff
@@ -0,0 +1,9 @@
+    width: 4.5rem;
+    transition: 1s;
+    
+    &:hover {
```
Thanks for animations :)

IvanCeban commented on sass/_content.scss:
```diff
@@ -0,0 +1,68 @@
+.content {
+    
```
Avoid adding empty lines inside on element. Finally it will be compiled in a minified file of course, so there is no impact on the product. But it is difficult to read such code.
Suggestion:
```
.content {
    padding: 2.5rem;
    display: grid;
    grid-template-columns: .23fr .44fr .33fr; 
    grid-gap: 1.1rem;
    grid-auto-flow: dense;
    
    @media(max-width: 992px) { ...
```

IvanCeban commented on sass/_content.scss:
```diff
@@ -0,0 +1,68 @@
+    }
+        
+    &--3 {
+        background-color: lighten($color-primary, 10%);
```
thanks for using scss functions

IvanCeban commented on sass/_content.scss:
```diff
@@ -0,0 +1,68 @@
+        grid-row: span 2;
+        
+        display: flex;
+        @include center-content-in-flex-item;
```
Thanks for using mixins!

IvanCeban commented on sass/_layout.scss:
```diff
@@ -0,0 +1,76 @@
+    display: grid;
+    grid-gap: .8rem;
+    
+    grid-template-rows: .07fr .86fr .07fr;
```
.07fr .86fr .07fr; - it's a pain )

if you want to use fr, you could just write: 1fr 12fr 1fr. You just need to keep proportions. 
If you want the fixed size, use "rem": 6rem 1fr 6rem; In this case you will have fixed size header and footer, but flexible content

IvanCeban commented on index.html:
```diff
@@ -0,0 +1,84 @@
+            </div>
+        </aside>
+        
+        <div class="content">
```
You could use tag "main" here

IvanCeban commented on sass/_layout.scss:
```diff
@@ -0,0 +1,76 @@
+.container {
+    background-color: $color-divider;
+    width: 100vw;
+    height: 900px; // later will be defined by the content
```
Instead of fixed height you could use min-height: 100vh;
In this case even if the content is not tall enough, your container will always have the height of the screen

IvanCeban commented on sass/_navigation.scss:
```diff
@@ -0,0 +1,156 @@
+            }
+        }
+                
+        &::before {
```
Thanks for nice animations!

IvanCeban commented on sass/_navigation.scss:
```diff
@@ -0,0 +1,156 @@
+    }
+}
+
+.side-navigation {
```
If you change the HTML part of your .side-navigation, you will have to rewrite css too.

IvanCeban reviewed (commented):
You did a great job!
Using npm, SASS, HTML5, BEM and lots of animations made me happy! Thank you!
There are some suggestions. Nothing critical, but good to take in consideration and to fix.

Commit:
Add div for container, change content div to main

Commit:
Change grid template fractions, remove empty lines

Commit:
Change side-navigation html and scss

Commit:
Change container min-height and content grid rows to adapt to it

oladelmar commented:
Thanks for your comments!)
I've made some changes. Please, have a look when you can.

IvanCeban commented:
Looks great! There left just .side-navigation refactoring. Are you going to do that?

oladelmar commented:
> Looks great! There left just .side-navigation refactoring. Are you going to do that?

I thought I already did that :) In the last but one ( 66a0850 ) commit.
I removed the .background div and applied the styles to the .side-navigation itself, moved the checkbox out of the .side-navigation and changed the css accordingly.
Or didn't I get you right?) Is there something else to change?

IvanCeban commented:
Oh, you are right)
I saw the `div class="sidebar__background">&nbsp;</div>` And decided that you didn't change the side-nav yet. Remove the background block as you don't use it anymore and you are done!

Commit:
Remove .sidebar__background div

oladelmar commented:
Thanks)
